src/h2n_fetcher.erl

```diff
@@ -585,7 +585,7 @@ take_at_least_one_second(Thunk) ->
     Result = Thunk(),
     Now = erlang:system_time(milli_seconds),
     Wait = Now - Start,
-    case Wait > 1000 of
+    case Wait > 500 of
```

Comment:
Github won't let me comment there, but i think the timer:sleep(1000 - Wait) should also be updated.